tezos/interop/src/ffi.rs

```diff
@@ -1,100 +1,113 @@
 // Copyright (c) SimpleStaking and Tezedge Contributors
 // SPDX-License-Identifier: MIT
-
-use ocaml::{List, Str, ToValue, Tuple, Value};
+use znfe::{
+    ocaml_alloc, ocaml_call, ocaml_frame, IntoRust, OCaml, OCamlBytes, OCamlInt32, OCamlList, ToOCaml,
+    OCamlFn1
+};
 
 use tezos_api::ffi::*;
 use tezos_api::identity::Identity;
 
 use crate::runtime;
 use crate::runtime::OcamlError;
 
-pub type OcamlBytes = Str;
-pub type OcamlHash = OcamlBytes;
-
-pub trait Interchange<T> {
-    fn convert_to(&self) -> T;
-    fn is_empty(&self) -> bool;
-}
-
-fn copy_to_ocaml(bytes: &[u8]) -> Str {
-    let mut arr = Str::new(bytes.len());
-    let data = arr.data_mut();
-    for (n, i) in bytes.iter().enumerate() {
-        data[n] = *i;
+mod tezos_ffi {
+    use znfe::{ocaml, Intnat, OCamlBytes, OCamlInt32, OCamlList};
+    ocaml! {
```

Comment:
@tizoc Bruno, I am thinking that this:
```
// TODO: probably needs to be moved to a better place
unsafe { tezos_interop_callback::initialize_callbacks() };
```
should go somewhere here, for example: https://doc.rust-lang.org/beta/std/sync/struct.Once.html "Useful for one-time initialization for FFI or related functionality" :D

it makes sense, because `mod tezos_ffi` defines ffi interface to ocaml and this calls needs those callbacks,
so everything will be defined at one place and tied together,

and maybe runtime.rs initialize_environment() should be also called here, maybe without `lazy_static OCAML_ENV`, runtime.rs will be more reusable,
or maybe some global tezos_interop::ffi_init() function will be much better then this lazy_static,

what do you think?

Reply:
I moved the initialization code around in https://github.com/simplestaking/tezedge/pull/261/commits/6ac36a9885ab1df2cae0ab9b9082dfe2161e0a2e

I'm not sure about how to go around the `lazy_static OCAML_ENV` change, we still need that handle to be a singleton, because we can only have a single OCaml runtime per process. Can you expand a bit on what would be the change there?